llvm/docs/GlobalISel/MIRPatterns.rst

```diff
@@ -514,3 +514,21 @@ of operands.
     (match (does_not_bind $tmp, $x)
            (G_MUL $dst, $x, $tmp)),
     (apply (COPY $dst, $x))>;
+
+
+ Gallery
+----------------
+
+We should use precise patterns that state our intentions. Please avoid
+using wip_match_opcode in patterns.
+
+.. code-block:: text
+  :caption: Example fold ext(trunc)
+
+(match (wip_match_opcode G_ZEXT):$root,
+
+(match (G_TRUNC $src, $x),
+       (G_ZEXT $root, $src),
+
+(match (G_TRUNC $src, $x, (MIFlags NoUWrap)),
+       (G_ZEXT $root, $src),
```

Comment:
Not sure why this is showing 2 nearly identical patterns. What is the MIFlags doing if the first pattern didn't need flag checks?

Reply:
The third pattern is more precise. The one above hits with no flags and with NoSWrap.

Reply:
It's more precise, but if the first pattern works you wouldn't write the second

Reply:
There is a lot of misbehaviour in Combine.td. I just want some examples that we can point to in review.